### PR TITLE
Template Registry - Remove redundant entries

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

While discussing with @aklkv he told explained me that there are redundant entries in the `template-registry.ts` file, in particular all the ones that have this syntax: `HdsXyz: typeof HdsXyzComponent`. The reason is that the `template-registry.ts` is used only to determine types/signatures of components used in the **_templates_** (HBS) and this means there's no use for such pattern (`HdsXyz`). In the `gjs/gts` files the components are imported so their signature are explicit.

### :hammer_and_wrench: Detailed description

In this PR I have:
- removed redundant entries in the template registry

***

### 👀 Component checklist

- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by alex-ju — COMMENTED** [review]
We introduced these signatures [at HCPs request](https://hashicorp.slack.com/archives/C7KTUHNUS/p1699295276662279) 😕

**Comment by didoo** [comment]
> We introduced these signatures [at HCP's request](https://hashicorp.slack.com/archives/C7KTUHNUS/p1699295276662279) 😕

@aklkv I've mentioned you in the [slack conversation](https://hashicorp.slack.com/archives/C7KTUHNUS/p1716368123018029?thread_ts=1699295276.662279&cid=C7KTUHNUS) but for future reference, can you also answer here: are these PascalCase entry actually needed or not?

**Comment by aklkv** [comment]
> > We introduced these signatures [at HCP's request](https://hashicorp.slack.com/archives/C7KTUHNUS/p1699295276662279) 😕
> 
> @aklkv I've mentioned you in the [slack conversation](https://hashicorp.slack.com/archives/C7KTUHNUS/p1716368123018029?thread_ts=1699295276.662279&cid=C7KTUHNUS) but for future reference, can you also answer here: are these PascalCase entry actually needed or not?

Certainly, I think I might have prompted this request 6 months ago with out having proper knowledge of how things work, I am hoping today I better equipped, but correct me if I am wrong.
first of template-registry  file is only needed for consumer that are looking to add glint  but still want to use .hbs  file, with that in mind there are only two ways we can call components from template:

```js
'Hds::Alert': typeof HdsAlertIndexComponent;
'hds/alert': typeof HdsAlertIndexComponent;
```

to summarize template-registry needs following formats:

```js
 export default interface HdsComponentsRegistry { 
  // Components
  'My::Component': typeof MyComponent,
  'my/component': typeof MyComponent,

  // Helpers
  'my-helper': typeof MyHelper,

  // Modifiers
  'my-modifier': typeof MyModifier,
}
```

**Comment by didoo** [comment]
@aklkv thank you so much for the clarification!

**Review by alex-ju — APPROVED** [review]
